### PR TITLE
Publish package on PyPI

### LINKED ISSUE #4: PyPI Upload of Python Bindings
Able to successfully upload and then install python bindings to test.pypi:
Steps taken:
```
cd /path/to/python/bindings/
python3 setup.py sdist 
```
*NOTE*: test.pypi does not accept linx .whl files
```
python3 -m twine upload --repository-url https://test.pypi.org/legacy/ dist/*
```
*NOTE*: dist folder will be automatically created, install twine as necessary
```
python3 -m pip install --index-url https://test.pypi.org/simple/ --no-deps precice
```
*NOTE*: test.pypi and pypi do not allow repititions of package name + revision. Even if revision is removed or deleted.

NOTE: Please `update pip, wheel, setuptools` prior to installation. Otherwise things break with no clear errors.

### PR DESCRIPTION
**Pull request by BenjaminRueth**
closes #4 

:warning: note that this PR removes all MPI related commands from `setup.py` :warning: 

# Todo

- [x] merge https://github.com/precice/python-bindings/pull/36 first
- [x] ~~also publish wheels? No ->~~ Postponed to https://github.com/precice/python-bindings/issues/37
- [x] ~~publish on release to PyPI, not PyPI test~~ Postponed to https://github.com/precice/python-bindings/issues/39, since we want to release and we are waiting for https://github.com/pypa/pypi-support/issues/189.
- [x] Use:
    ```
    on:
      push:
        branches:
          - master
    ```
    not
https://github.com/precice/python-bindings/blob/ff439ef7fe4384c9faf0f434be5e95eb909f7eff/.github/workflows/pythonpublish.yml#L3
- [x] Implement a proper versioning strategy #31

### CODE REVIEW

**Comment by BenjaminRueth** [comment]
Note: We currently cannot access our PyPI.org account. There is already a user `precice` that is linked to our email address. However, resetting the password does not work. I already opened a password reset request over the support (https://github.com/pypa/pypi-support/issues/189).

File: .github/workflows/pythonpublish.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: Upload Python Package
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
```
Comment (fsimonis) [inline .github/workflows/pythonpublish.yml]: Maybe uploading on tag would prevent mistakes?

https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-branches-and-tags

File: setup.py [hunk 2]
```diff
@@ -17,57 +17,17 @@
 
 # name of Interfacing API
 APPNAME = "pyprecice"
-APPVERSION = "2.0.0a1"  # todo: should be replaced with precice.get_version() as soon as it exists , see https://github.com/precice/precice/issues/261
+APPVERSION = "2.0.0a4"  # todo: should be replaced with precice.get_version() as soon as it exists , see https://github.com/precice/precice/issues/261
```
Comment (fsimonis) [inline setup.py]: Let's wait until #31 is resolved. :wink:

File: .github/workflows/pythonpublish.yml (hunk 1 above)
Comment (BenjaminRueth) [reply .github/workflows/pythonpublish.yml]: Done 497ee70

File: solverdummy/solverdummy_parallel.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+solver_process_size = 1
+
+interface = precice.Interface(participant_name, configuration_file_name, MPI.COMM_WORLD.Get_rank(), MPI.COMM_WORLD.Get_size())
+    
+mesh_id = interface.get_mesh_id(mesh_name)
+
+dimensions = interface.get_dimensions()
+vertices = np.ones((n, dimensions)) * MPI.COMM_WORLD.Get_rank()
+
+data_indices = interface.set_mesh_vertices(mesh_id, vertices)
```
Comment (uekerman) [inline solverdummy/solverdummy_parallel.py]:
```suggestion
vertex_ids = interface.set_mesh_vertices(mesh_id, vertices)
```

File: setup.py (hunk 2 above)
Comment (BenjaminRueth) [reply setup.py]: #31 is resolved now.

**Pull request merged**